src/pages/companies/company-list.component.js

```diff
@@ -8,32 +9,17 @@ import { useHistory } from 'react-router-dom';
 import RoleFilter from 'components/role-filter/role-filter.component';
 
 const CompaniesList = () => {
-  const [companies, setCompanies] = useState([]);
-  const [loading, setLoading] = useState(false);
   const history = useHistory();
 
-  const onStart = () => {
-    setLoading(true);
-  };
-
-  const onSuccess = (data) => {
-    setCompanies(data);
-  };
-
-  const onFailure = () => {
-    toast.error('Erro ao carregar empresas!');
-  };
-
-  const onCompleted = () => {
-    setLoading(false);
-  };
-
-  useEffect(() => {
-    getCompanies({ onSuccess, onStart, onFailure, onCompleted });
-  }, []);
+  const { data, isLoading, isError } = useQuery('companies', getCompanies);
+  const companies = data?.data || [];
 
   const render = () => {
-    if (loading)
+    if (isError) {
+      toast.error('Erro ao carregar empresas!');
```

Comment:
Instead of showing a toast here, I think we should render a message on the page. Could you do that?

Reply:
Sure, I'll look into that.

Reply:
Can you attach a screenshot so we can see what it looks like?

Reply:
Sure, after the request has failed, this is what it looks like:
![image](https://user-images.githubusercontent.com/79973491/163481775-af77aea1-de09-4f2e-8882-428236f07962.png)